### DISCUSSION TITLE
Improvements in Terminal for native "Windows PowerShell"

### DISCUSSION THREAD

**Discussion by rrangelo** [apache/netbeans#5789] [Ideas]
Hello! I am a Netbeans user for 15 years, I always use it in Windows.

Other IDE's, and even code editors, have a native integration into their terminal tool, connecting without the need to install additional software in Windows, such as: Visual Studio Code and Jetbrains Intellij IDEA (in any of its editions).

I would love to be able to use the terminal in the same way as those tools mentioned above do.

-- For Spanish speakers / Para hablantes de español ---

¡Hola! Soy usuario de NetBeans desde hace 15 años, siempre lo uso en Windows.

Otros IDE's, y hasta editores de código, cuentan con una integración nativa en su herramienta de terminal, conectándose sin la necesidad de instalar software adicional en Windows, tales como: Visual Studio Code y JetBrains IntelliJ IDEA (en cualquiera de sus ediciones).

Me encantaría poder usar el terminal de la misma manera como lo hacen esas herramientas mencionadas anteriormente.

**Comment by Chris2011** [comment]
Hey, I feel your pain and I already had a look into the core and I gave up first. I tried to just comment out the check for windows to show the dialog about the missing support but this was somehow not enough. So I think it is not thaaaat easy to implement it. Maybe it would be possible to support modern WSL which would be a nice improvement but I'm also want to PowerShell and CMD. There is also a plugin called JConsole: https://github.com/fungl164/JConsole but it buggy in some specific cases and not maintained anymore so maybe the bugs are not that hard to fix but needs to check it. I also want to improve the UI, but not the IO communication.

**Reply by rrangelo** [reply]
Hi friend!

I found that lib [JLine](https://github.com/jline/jline3). It looks a great option to improve the Apache NetBeans Terminal. I hope it can be help you.

**Reply by Chris2011** [reply]
Hey, thx we already talked about it here: https://github.com/apache/netbeans/issues/6210 maybe the pty4j is good to go.
